staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances.go

```diff
@@ -316,10 +326,17 @@ func (az *Cloud) InstanceType(ctx context.Context, name types.NodeName) (string,
 		if err != nil {
 			return "", err
 		}
-		if isLocalInstance {
-			if metadata.Compute.VMSize != "" {
-				return metadata.Compute.VMSize, nil
+		if !isLocalInstance {
+			if az.vmSet != nil {
```

Comment:
why do this code change? not use `metadata.Compute.VMSize, nil` directly?

Reply:
when this is not local instance, and credentials not provided, we should report errors.

the original logic has been moved to L331.